app/qml/map/components/MMMeasureCrosshair.qml

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+import Qt5Compat.GraphicalEffects
+
+import mm 1.0 as MM
+import "../../components"
+
+Item {
+    id: root
+
+    required property string text
+    /*required*/ property var qgsProject
+    /*required*/ property var mapSettings
+
+    property bool canCloseShape: false
+    property bool shouldUseSnapping: false
+
+    property point center: Qt.point( root.width / 2, root.height / 2 )
+
+    property var recordPoint: snapUtils.recordPoint
+
+    property point screenPoint: snapUtils.snapped && __activeLayer.vectorLayer ? __inputUtils.transformPointToScreenCoordinates(__activeLayer.vectorLayer.crs, mapSettings, recordPoint) : center
+
+    property real outerSize: 60 * __dp
+    property real innerDotSize: 10 * __dp
+
+    implicitWidth: row.width
+    implicitHeight: __style.mapItemHeight
+
+    signal closeShapeClicked
+
+    MM.SnapUtils {
+      id: snapUtils
+
+      centerPosition: root.center
+      mapSettings: root.mapSettings
+      qgsProject: root.qgsProject
+      useSnapping: root.shouldUseSnapping
+      destinationLayer: __activeLayer.vectorLayer
```

Comment:
Ah cool, so we support snapping?

Reply:
By the way, `MMCrosshair` is creating its own `SnapUtils`, do we need to duplicate it here?